bindings/swift/xcode/IotaWallet/IotaWalletTests/await.swift

```diff
@@ -16,12 +22,59 @@ class SwiftAwait: XCTestCase {
         // Put teardown code here. This method is called after the invocation of each test method in the class.
     }
     
-    func testAsyncAwait() async throws {
-        let wallet = IotaWallet.Wallet()!
+    func testCreateAccount() async throws {
+        let wallet = try! IotaWallet.Wallet()
+        
         let request = "{\"cmd\": \"CreateAccount\", \"payload\": { \"clientOptions\": { \"node\": \"https://nodes.devnet.iota.org:443\" } }, \"signerType\": { \"type\": \"Stronghold\" } }";
         
         let response = try! await wallet.sendMessage(request)
-        print("\(#function) -> \(response)")
+        print("\(response)")
+    }
+    
+    func testManagerOptions() async throws {
+        let signer = #"{"Mnemonic":"acoustic trophy damage hint search taste love bicycle foster cradle brown govern endless depend situate athlete pudding blame question genius transfer van random vast"}"#
+        let client_options = """
+        {
+            "nodes":[
+                {
+                    "url":"https://localhost",
+                    "auth":null,
+                    "disabled":false
+                }
+            ],
+            "localPow":true,
+            "apiTimeout":{
+                "secs":20,
+                "nanos":0
+            }
+        }
+        """
+        
+        let manager_options = ManagerOptions(storageFolder: "teststorage", clientOptions: client_options, signer: signer)
+        
+        let json = try JSONEncoder().encode(manager_options)
+        let manager_options_json = String(data: json, encoding: .utf8)!
+        
+        let _ = try! IotaWallet.Wallet(managerOptions: manager_options_json)
+    }
+    
+    func testEvents() async throws {
+        let wallet = try! IotaWallet.Wallet()
+        let expectation = self.expectation(description: "TestEvents")
+
+        try! wallet.listen([]) { (message: String?, error: Error?) in
+            print("Event received: \(message!)")
+            expectation.fulfill()
+        }
+        
+        // TODO: Create account and do something that would trigger an event
```

Comment:
Do you think it would be good to have a test event specially for such a case behind  a feature flag, so we don't have to do interactions that create an event, but can just call `emit_test_event()`?

Reply:
Yeah, but I wouldn't know where such a function would fit.

The function would ideally be in wallet.rs, but calling it from here in the Swift would mean there has to be a binding that's only used for testing, maybe such function should only exist in debug mode to avoid misuse in production.

What do you think?

Reply:
Is it possible to only be available in debug mode? Definitely behind a feature, but can't do any harm anyways

> Yeah, but I wouldn't know where such a function would fit.

I would just add it to the bottom of the AccountManager, the `verify_integrity()` method also only makes sense during development and should probably behind the same feature then

Reply:
> Is it possible to only be available in debug mode? Definitely behind a feature, but can't do any harm anyways

I found this https://doc.rust-lang.org/reference/conditional-compilation.html#debug_assertions, but maybe it's fine putting it behind a feature.

> the verify_integrity() method also only makes sense during development and should probably behind the same feature then

Good point, I'll handle that as well.

Reply:
Since Rust features don't translate to other languages, I eventually decided to go with making emit_test_event() available in debug builds only, I think it enforces the constraint that this should only be used while testing rather than in production.